lua/telescope/actions/init.lua

```diff
@@ -906,13 +906,19 @@ local entry_to_qf = function(entry)
     end
   end
 
-  return {
+  local qf_entry = {
     bufnr = entry.bufnr,
     filename = from_entry.path(entry, false, false),
     lnum = vim.F.if_nil(entry.lnum, 1),
     col = vim.F.if_nil(entry.col, 1),
     text = text,
   }
+
+  if entry.type then
+    qf_entry.type = entry.type
```

Comment:
As [`:h setqflist()`](https://github.com/neovim/neovim/blob/38bb0e1da868fd7593387319376d0e167616638b/runtime/doc/builtin.txt#L6683) metions, the `type` is a single character, maybe it should be the first letter of `entry.type`.

Reply:
[This](https://github.com/neovim/neovim/blob/38bb0e1da868fd7593387319376d0e167616638b/runtime/lua/vim/diagnostic.lua#L1987) is how the `neovim core` handles the `type` field, the distinction between the two lies in mapping the `Hint` to `N`.

Reply:
yeah we should handle it the same way although i think this should be part of `builtin/__diagnostic`, the entry that is being produced there. because its specific to diagnostics and not to the mapping from telescope entry -> qf list entry.

So we can add a new entry there, like hint which is already single line and correctly mapped. does that make sense?

Reply:
> yeah we should handle it the same way although i think this should be part of `builtin/__diagnostic`, the entry that is being produced there. because its specific to diagnostics and not to the mapping from telescope entry -> qf list entry.
> 
I couldn't agree more.

I took a quick look at the [code](https://github.com/gh-liu/telescope.nvim/blob/4bd510943836edcf588c4dedff814dbaf372a273/lua/telescope/make_entry.lua#L1121), and noticed that the `entry.type` is used in `display` and `complete_tag`, or possibly elsewhere, I am wondering if the `entry.type` should be align with the `type` field of `qf entry`. If not, maybe a new field could be added for this purpose, such as `qf_type`.

Reply:
i prefer a new key that that `entry.qf_type` is good enough

Reply:
Done.